### PR TITLE
Components: Start adding types progressively

### PR DESCRIPTION
**Pull request by sirreal**
## Description

Add types to the `Dashicon` leaf component in `packages/components/`.

Part of #18838.
## How has this been tested?
```
npm run build:package-types
```

Verify that this passes, and that types are being built in `packages/components/build-types/dashicon/`.

## Screenshots
N/A

## Types of changes
Add types

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/components/src/dashicon/index.js [hunk 1]
```diff
@@ -11,6 +24,10 @@ function Dashicon( { icon, size = 20, className, ...extraProps } ) {
 	return (
 		<span
 			className={ iconClass }
+			// Ignore reason: span attributes are the global attribtues which do not include width/height
```
Comment (ockham) [inline packages/components/src/dashicon/index.js:27]:
```suggestion
			// Ignore reason: span attributes are the global attributes which do not include width/height
```

File: packages/components/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+	"extends": "../../tsconfig.base.json",
+	"compilerOptions": {
+		"rootDir": "src",
+		"declarationDir": "build-types",
+	},
+	"references": [],
+    "include": [
+		"src/dashicon/*"
+	],
+}
```
Comment (ZebulanStanphill) [inline packages/components/tsconfig.json:11]: This is missing a newline.

Comment (ockham) [reply packages/components/tsconfig.json]: 451b9b56f8

**Review by ockham — APPROVED** [review]
LGTM (pending CI -- I had some trouble trying to restart them in bulk).

**Pull request merged**